AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/EndActivity.java

```diff
@@ -37,6 +38,28 @@ protected void onCreate(final Bundle savedInstanceState) {
         });
 
         endTextView = findViewById(R.id.end_return);
-        endTextView.setOnClickListener(l -> finish());
+        endTextView.setOnClickListener(l -> {
+            startIntroActivity();
+            finish();
+        });
+    }
+
+    private void startIntroActivity() {
+        final Intent intent = new Intent(this, IntroActivity.class);
+        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        startActivity(intent);
+    }
+
+    @Override
+    public void onBackPressed() {
+        super.onBackPressed();
+        startIntroActivity();
+    }
+
+    @Override
+    protected void onDestroy() {
+        Log.d(LOG_TAG, "EndActivity - onDestroy");
+        super.onDestroy();
```

Comment:
These appear to be changes that are not directly related to Teams Interop. Should they be applied to the main branch first?